### DISCUSSION TITLE
Personal Observations on Znote

### DISCUSSION THREAD

**Discussion by rbharathkumar** [alagrede/znote-app#52] [Ideas]
Hi,
Today tried out the Znote app after hearing some good things about the editor.
The editor was very fun to use and explore. Great work.

Below are some of the observations, I had. Please note that all these are personal and maybe wrong or does not align with the app's vision. They can ignored if not valid.

Thank you for all the hard work.

- Downloaded the release from GitHub releases, it prompted me for update to the lab version. Would be nice to document the difference which I understand is the time gap between the actual release and the release in the respective stores ?
- An About section of the Editor would be nice to have. It would be helpful in case we wanted to check the currently installed version number or link to the github page.
- Information on the location and format of the save file would be nice to have.
- Roadmap of features to be implemented/currently worked upon. This would attract the users. Of course this bears no compulsion on the developer to deliver them. Just a good to know item.
- License information.
- Option to modify the fonts.
- Option to enable Line Numbers.
- Option to add a table along with its related options such as adding/moving rows, columns.
- On the toolbar, tool tips are available only for the first three options. Would be nice to have for all the items. This would be really helpful in case of the panel options which I thought was for Justification of text.
- The Horizontal Line Button HR - Though it is clear represents Horizontal Line, I had mistook it for a Header level, #, ## which I later found is the `Titl` Button
- Auto-Generation on pressing Enter. Say we create a list in the Editor, would be nice to have it incremented instead of typing manually.
- Alternatively on a new line, would be nice to have a pop-up suggesting the Markdown element to be added or maybe a trigger character like / in Notion/Obsidian which would in turn have that option
- Option to add TOC at top of the document. Already seen in some issues.
- The Outline Feature can be added as an optional side bar that does not block the preview window. The Outline can also be presented a bit better with proper indent. Currently has Header Tags in it.
- The option "<" to Hide the "Note List" is available only when the "Tags Bar" has been hidden, meaning we need two clicks to have only the editor and preview windows alone.
- Option for "ACTIVATE" is available along with certain options available behind the PRO option - Does this mean the editor is soon to become a Paid Editor ?

**Comment by alagrede** [comment]
Hi @rbharathkumar,

Thank you very much for your feedback 🙏. I'm agree with you, I should bring a clearer vision of the roadmap.
I will also add some of your interesting suggestions in my next releases 😊

To answer to your question about the future of Znote. I provide free version of Znote on Apple and Microsoft store and it will be free forever! ✓
You're right, there is a time gap between stores and Github because there are more constraints to deploy small incremental versions on stores. 
I also provide a more complex version of Znote, **the Znote Lab** which is also a valuable Javascript playground in addition to the note taking application (see website https://znote.io/). This Lab version contains all free features as well as PRO features for advanced use cases.

Today I have too many versions to maintain and I'm trying to rationalize that to be more efficient.
I recently started a campaign with this small prompt to encourage users to upgrade to the lab version which is always the latest version and contains all possible features. Buy a license for the PRO version is the only way for me to get some support from users allowing me to actively continue this application. Once this campaign done, I will stop the light (more updated) Github version of Znote to focus exclusively on the Stores free version and the Lab (with PRO) more updated. (And without intrusive prompt to promote the Lab 😅).

I hope you understand my point of view.
We stay in touch for your suggestions.

Thank you
Anthony

**Reply by rbharathkumar** [reply]
Thank you very much for your response and for considering my suggestions.

Just wanted to see if my understanding is correct.

| Version   | Download From                                | Version          | Pro Features | Cost                           |
| --------- | -------------------------------------------- | ---------------- | ------------ | ------------------------------ |
| znote     | Github                                       | Latest           | No           | Free                           |
| znote-lab | Website                                      | Latest           | Yes          | License to unlock Pro Features |
| znote     | App Stores/<br/>Other means of distributions | Slightly Delayed | No           | Free                           |

My thoughts on this.

1. You can try maintaining a single app with the JS Playground disabled by default. Unlocking it would require the user to buy the license. Not sure of the technical challenges, but this would mean that people from all backgrounds, not just JS can continue to use the app for its markdown capabilities.
2. I hope common editing and export features such as export to pdf are available to all users and not hidden behind PRO features. Features such as Locked Folders, Extra Sync Capabilities to Cloud Storage, Templates, Themes and especially the playground are some examples of PRO features.
3. I hope that you will continue to publish versions in GitHub even if they are just the same ones as the Website (Apart from the Store Versions). This will ensure that users will continue to download from GitHub, understand the changes in the release notes and file issues or have discussions. If the downloads are going to be limited to the website/ Store then this useful channel of communicating with the user will be lost.

As always it is just a common user's observation with no understanding of the technical effort or cost considerations. So please feel free to ignore if not valid.

All the best.

**Comment by alagrede** [comment]
Well summarized.
1. The Lab version will be merged with the Github one. My medium-term plan is indeed to hide a bit the JS playground and allow it to be manually enabled and only then ask to purchase a license for the advanced features (playground + more cloud space). My current prompt for the Lab version is temporary and a bit clumsy.
2. I agree to open the export options by default. It was not planned but I don't see any inconvenience. 
3. Don't worry, I'll never go down this Github space. I like to chat with all my users and try to make this app better and better 😊. Future releases will be placed in this Github releases section. 

Thank you.

**Reply by rbharathkumar** [reply]
Thank you for your response...
Looking forward to future developments on the editor...
All the Best...

**Comment by rbharathkumar** [comment]
Thank you for releasing a new version of znote and thank you for considering some of the features suggested.

The collapsible blocks is a great touch. Not sure if it was always available...

Please consider some more observations apart fromt he ones I have already listed in the first discussion...

1. For a change tried the znote AppImage on a plasma desktop. The window controls such as Minimize and Maximize and close were not available. Not sure if this is a global issue - maybe try on a linux vm.
2. Consider making some changes/standardizations on the right click menu.
![image](https://user-images.githubusercontent.com/38585383/154841512-6d40ccdc-f48e-47e4-b816-d85b15fe86ae.png)
2.1. By Default on a new note, we have options for Markdown tables, decode, encode and command pallete. 
2.2. Cut and Copy can be disabled (I mean greyed out) and activated when content is available.
2.3. Paste Option is not available.
2.4. Consider moving the options inside a sub menu say
        Insert -> Line, code etc
        Actions -> Tables -> Table Actions
        Actions -> Format -> Formating Options
        Actions -> Base64 -> Encode/Decode
3. The outline is much better - Maybe can have collapsible headings and the link kind of line below headings can be avoided - Also on shift of focus the outline panel can be closed instead of hitting the close button manually - Purely cosmetic and personal choice..
4. The list icon usually has three different kind of symbols showing each level of indent say shaded circle, unshaded circle and a shaded square. Currently all three have the same shaded circle. Please refer to how github changes the below items.
```
* Point 01
    * Sub Point 01
        * Sub Sub Point 01
```

* Point 01
    * Sub Point 01
        * Sub Sub Point 01
5. Similarly for ordered list on the editor side, the first and second levels are highlighted. Any level below that is not highlighted.
```
1. Ordered List 01
    1.1. Ordered List 02
        1.1.1. Ordered List 03
            1.1.1.1. Ordered List 04
```
6. Tables
6.1. Can be added as an option in toolbar.
6.2. Consider changing name from CSV to Markdown table to insert table since there is no CSV to begin with.
6.3. The Table insert can consider asking for the number of rows and columns to insert.
6.4. Option for inserting/deleting/re-arranging rows/columns can be provided.
6.5. Consider disabling/greying out the option Markdown table to CSV if no markdown table is selected.

7. Is it possible to currently paste an image into the editor ? For some reason I could not do it.

8. Consider changing the command pallete shortcut to the standard - Cmd+Shift+P as in many editors.

9. The trigger character such as `/` would be very useful to insert new elements.

Thanks for all the hard work. 
All the Best....

**Comment by alagrede** [comment]
Hi @rbharathkumar,
Waouh, thank again for your interest and feedback! 😊
I can't implement all your demands but this is really great to have your feedback.
I will start to organize all requests into a dashboard to share advancement and future features (also like suggested)

FI: instead of `/` try `Cmd+space`

Best regards,
Anthony

**Reply by rbharathkumar** [reply]
Hi @alagrede,
Thank you very much for your response...
The above points are merely observations and not demands...
Apologies if the note had sounded like a demand....
As I have mentioned earlier, please do not be compelled to implement them - was just trying to help...

**Reply by alagrede** [reply]
Yes yes of course 👍. Thank you again.
